agenda/data/src/main/java/com/gkp/agenda/data/OfflineAgendaRepository.kt

```diff
@@ -177,9 +186,50 @@ class OfflineAgendaRepository(
         return localAgendaDataSource.getAgendaItemById(id)
     }
 
+    override suspend fun pushOfflineAgendaItems() {
+        val userId = sessionStorage.getAuthInfo().userId
+        val offlineDeletedAgendaItems = deletedAgendaItemsDao.getAllDeletedAgendaItemsForUser(userId)
+
+        val deletedJobs = offlineDeletedAgendaItems.map { deletedAgendaItem ->
+            scope.launch {
+                syncAgendaItems.syncDeletedAgendaItem(
+                    agendaItemId = deletedAgendaItem.id,
+                    agendaItemType = AgendaItemType.valueOf(deletedAgendaItem.agendaItemType)
+                )
+            }
+        }
+
+        val offlineUpdatedAgendaItems = updatedAgendaItemsDao.getUpdatedItemsByUserId(userId)
+        val updatedJobs = offlineUpdatedAgendaItems.map { updatedAgendaItem ->
+            scope.launch {
+                syncAgendaItems.syncUpdatedAgendaItem(
+                    agendaItemId = updatedAgendaItem.id,
+                )
+            }
+        }
+
+        val offlineCreatedAgendaItems = createdAgendaItemsDao.getCreatedAgendaItemsByUserId(userId)
+        val createdJobs = offlineCreatedAgendaItems.map { createdAgendaItem ->
+            scope.launch {
+                syncAgendaItems.syncCreatedAgendaItem(
+                    agendaItemId = createdAgendaItem.id,
+                )
+            }
+        }
+
+        updatedJobs.joinAll()
+        createdJobs.joinAll()
+        deletedJobs.joinAll()
+    }
+
+    override suspend fun periodicalSyncAgendaItems() {
+        syncAgendaItems.syncFullAgenda()
+    }
+
     override fun logout() {
         networkApiCall {
             sessionStorage.resetAuthInfo()
+            syncAgendaItems.cancelWorkers()
             taskyRetrofitApi.logout()
         }.launchIn(scope)
```

Comment:
Only the last call here is a network call right?

Could something like this make more sense?

```
private fun logout() {
  applicationScope.launch {
    // this will cancel the workers
    syncAgendaItemsScheduler.cancelAllSyncs()

    // this will delete all the user’s agenda items
    agendaRepository.deleteAllAgendaItems()
    
    // this will call the logout API (you can use yours here)
    authCoreService.logout()

    // here we clear the user’s session
    sessionStorage.setSession(null)
  }
}
```